oci/runtime_v2.go

```diff
@@ -214,6 +245,90 @@ func (r *RuntimeV2) ExecSyncContainer(c *Container, command []string, timeout in
 	return &ExecSyncResponse{}, nil
 }
 
+func (r *RuntimeV2) execContainer(c *Container, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool) (int32, error) {
+
+	logrus.Debug("Runtimev2.execContainer() start")
+	defer logrus.Debug("RuntimeV2.execContainer() end")
+
+	// Create IO fifos
+	execIO, err := cio.NewExecIO(c.ID(), fifoGlobalDir, tty, stdin != nil)
+	if err != nil {
+		return -1, errdefs.FromGRPC(err)
+	}
+
+	execIO.Attach(cio.AttachOptions{
+		Stdin:     stdin,
+		Stdout:    stdout,
+		Stderr:    stderr,
+		Tty:       tty,
+		StdinOnce: true,
+		CloseStdin: func() error {
+			return nil
+		},
+	})
+
+	pSpec := c.Spec().Process
+	pSpec.Args = cmd
+
+	any, err := typeurl.MarshalAny(pSpec)
+	if err != nil {
+		return -1, errdefs.FromGRPC(err)
+	}
+
+	// Generate a unique execID
+	execID := generateID()
+
+	request := &task.ExecProcessRequest{
+		ID:       c.ID(),
+		ExecID:   execID,
+		Stdin:    execIO.Config().Stdin,
+		Stdout:   execIO.Config().Stdout,
+		Stderr:   execIO.Config().Stderr,
+		Terminal: execIO.Config().Terminal,
+		Spec:     any,
+	}
+
+	// Create the "exec" process
+	if _, err := r.task.Exec(r.ctx, request); err != nil {
+		return -1, errdefs.FromGRPC(err)
+	}
+
+	// Start the process
+	if _, err := r.task.Start(r.ctx, &task.StartRequest{
+		ID:     c.ID(),
+		ExecID: execID,
+	}); err != nil {
```

Comment:
If start exec failed here, do we need to do cleanup such as delete the exec?

Reply:
Yes the cleanup would be nice, this is the last step after everything will be cleanly implemented.
And yes a call into `defer execIO.Close()` would be useful.

Reply:
`execIO.Close()` added.

Reply:
comment addressed.